### DISCUSSION TITLE
cbor, bson etc.

### DISCUSSION THREAD

**Discussion by AndreyAlifanov** [danielaparker/jsoncons#294] [Q&A]
Hi, Daniel!

Do binary formats parsers work?

I have very simple JSON and encoded it with jsoncons into cbor, bson, ubjson, msgpack.
After that I try to decode these files. And get various errors. Only cbor and ubjson work.

On complex JSON I can't read these formats at all.

**Comment by danielaparker** [comment]
Judging from feedback from users, all of the binary extensions are used. CBOR is undoubtable the most used. All have test coverage.

If you experience a problem with a particular case, I would suggest to submit an issue.

**Comment by AndreyAlifanov** [comment]
ok, I'll submit an issue.

**Reply by danielaparker** [reply]
Any updates on this issue? Thanks.

**Reply by AndreyAlifanov** [reply]
Hi.
I'm sorry. It was my mistake. I've fixed it and all works fine.
Except msgpack, but this format is not relevant to me.

**Reply by danielaparker** [reply]
What was the issue you had with msgpack? Can you provide an example that fails?

**Reply by AndreyAlifanov** [reply]
I parse json file, save it to msgpack. And can't read this msgpack.
I get 
_Request for the length of an array, map or string returned a negative result at position 1576_

All other formats work well.

Test project in attachment.

[JsonTest.zip](https://github.com/danielaparker/jsoncons/files/5830438/JsonTest.zip)

**Reply by danielaparker** [reply]
Thanks for sending the failing msgpack example. You're right, there was an issue with decoding the length of a string that was triggered by one of the items in your test input. Fixed on master.
